### PR TITLE
#2559. Add augmenting types tests. Part 2

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Augmentation-libraries/augmenting_types_A02_t15.dart [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+/// @description Checks that it is a compile-time error if an augmenting type
+/// and corresponding type do not have all the same modifiers (final, sealed,
+/// mixin, etc). Test `base mixin` in the main library
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+import augment 'augmenting_types_A02_t15_lib.dart';
+
+mixin C {}
```
Comment (eernstg) [inline LanguageFeatures/Augmentation-libraries/augmenting_types_A02_t15.dart:19]:
```suggestion
base mixin C {}
```

**Review by eernstg — COMMENTED** [review]
Looks good, just one typo (I presume).

File: LanguageFeatures/Augmentation-libraries/augmenting_types_A02_t15.dart (hunk 1 above)
Comment (sgrekhov) [reply LanguageFeatures/Augmentation-libraries/augmenting_types_A02_t15.dart]: Thank you for noticing!

**Review by eernstg — COMMENTED** [review]
LGTM

**Review by eernstg — APPROVED** [review]

**Comment by eernstg** [comment]
Oh, one more thing that I missed: It should be 'augment library ...' rather than 'library augment ...'. Perhaps it hasn't been implemented, but that should happen soon anyway.

**Comment by sgrekhov** [comment]
I'll do that as part of https://github.com/dart-lang/co19/issues/2576

**Pull request merged**